### ISSUE TITLE
Generic Query Filtered Response

### ISSUE THREAD

**Issue by verygoodsoftwarenotvirus** [dinnerdonebetter/dinnerdonebetter#106]
old and busted:

```go
// RecipeList represents a list of recipes.
type RecipeList struct {
	_ struct{}

	Recipes []*Recipe `json:"recipes"`
	Pagination
}
```

new hotness:

```go
type QueryFilteredResult[T any] struct {
	Data []T `json:"data"`
	Pagination
}

// RecipeList represents a list of recipes.
RecipeList QueryFilteredResult[*Recipe]
```

active questions: does that being a pointer fuck me over in any meaningful way? A brief attempt at de-pointerizing just this one list caused the database tests to go from ~4s to ~22s, so I anticipate that's not the answer to my question.

**Comment by verygoodsoftwarenotvirus** [comment]
This can't be done yet because the linter simply cannot cope with generics

**Comment by verygoodsoftwarenotvirus** [comment]
now implemented, so we should track #425
